app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -424,23 +409,38 @@ protected function addShipping( $totals, &$cartSubmissionData, $shippingAddress,
     /**
      * Adds the calculated shipping to the Bolt order from the admin context
      *
-     * @param Mage_Sales_Model_Quote_Address_Total  $shippingTotal      calculated shipping totals
+     * @param Mage_Adminhtml_Block_Sales_Order_Create_Totals  $totalsBlock      totals block
      * @param array                                 $cartSubmissionData data to be sent to Bolt
      * @param Mage_Sales_Model_Quote_Address_Rate   $shippingRate       shipping rate meta data
      * @param array                                 $boltFormatAddress  shipping address in Bolt format
      * @param Mage_Sales_Model_Quote                $quote              the quote associated with the order
      *
      * @return int    The total shipping in cents
      */
-    protected function addShippingForAdmin( $shippingTotal, &$cartSubmissionData, $shippingRate, $boltFormatAddress, $quote ) {
-        $totalShipping = $shippingTotal ? (int) round($shippingTotal->getValue() * 100) : 0;
-        $cartSubmissionData['shipments'] = array(array(
-            'shipping_address' => $boltFormatAddress,
-            'tax_amount'       => 0,
-            'service'          => $shippingRate->getMethodTitle(),
-            'carrier'          => $shippingRate->getCarrierTitle(),
-            'cost'             => $totalShipping,
-        ));
+    protected function addShippingForAdmin($totalsBlock, &$cartSubmissionData, $shippingRate, $boltFormatAddress, $quote ) {
+        $totalShipping = 0;
+        if ($shippingRate){
+            /* @var Mage_Sales_Model_Quote_Address_Total $addressShippingTotal */
+            $addressShippingTotal =  $totalsBlock->getTotals()['shipping'];
+            $totalShipping = $addressShippingTotal ? (int) round($addressShippingTotal->getValue() * 100) : 0;
+            $cartSubmissionData['shipments'] = array(array(
+                'shipping_address' => $boltFormatAddress,
+                'tax_amount'       => 0,
+                'service'          => $shippingRate->getMethodTitle(),
+                'carrier'          => $shippingRate->getCarrierTitle(),
+                'cost'             => $totalShipping,
+            ));
+        }else{
+            if ($quote->isVirtual()) {
+                $cartSubmissionData['shipments'] = array(array(
+                    'tax_amount'       => 0,
```

Comment:
The relevant change is here, namely the removal of the `shipping_address` entry.